### DISCUSSION TITLE
weird behaviour in lazer after realm restore

### DISCUSSION THREAD

**Discussion by newtonGnu** [ppy/osu#20021] [Q&A] [answered]
### missing beatmap data
i've had some self-caused trouble with lazer because i **accidentally opened an older version** of lazer. (this happened because when updating lazer my system keeps every version of the AppImage and they are not named in a comprehensible way, in the future i will make sure to delete every old version after updating to avoid this)

i took a look at this discussion #17148 but in my case the backups didn't contain "corrupt" so i wasn't sure how to restore. i also tried multiple older versions one after another because i wasn't sure which was which.

in this discussion #19648 the solution was also to rename the corrupt realm file but again, i don't have one.

instead i tried renaming one of the backup realm files and now all beatmaps show up again but their content is missing (see error messages), example:
> 2022-08-29 18:42:43 [error]: Beatmap failed to load (file _map name_ .osu not found on disk at expected location 0/05/05e4c2840701f6941ab50ecd07a42f3514c79fce2173150681b2c4bb893efd26).
> 2022-08-29 18:42:43 [verbose]: Beatmap background failed to load (file Background.jpg not found on disk at expected location 6/6c/6cd28cb977a8954e91b5eff0a64a7157de2831f9f7d5c024213fbdca1784db37).

so the "meta data" is back with all map names, scores and collections but all the content (images, audio) is missing.
is there another file i need to restore besides _client.realm_ ?

context for the log files: some of the maps (~170) work fine because i reimported them from my old stable folder but in total there are a bit more than ~400 maps and i would really like to restore the ~250 missing maps from my previous working lazer install.

here are the log files:
[logs.zip](https://github.com/ppy/osu/files/9447388/logs.zip)
these are all the backups (i used the marked one)
![backups](https://user-images.githubusercontent.com/43142305/187280715-5c1e21ab-cda4-4869-a02f-b8fd1cfe20b3.png)
there are also some files in the root osu directory (the ones with _ underscore where copied by me)
![root](https://user-images.githubusercontent.com/43142305/187281396-5624eaa8-7aba-4bc6-baac-2fbafe891ffb.png)

___

### broken update notification
side-note: got a notification to "downgrade" ? (i'm on 821 and got this message)
> 2022-08-29 18:42:28 [verbose]: ⚠️ A newer release of osu! has been found (2022.821.0 → 2022.816.0).

**Comment by peppy** [comment, marked as answer]
If files are missing then there is no way to recover. It means the files were deleted from the file store. You will need to reimport/redownload beatmaps.

> 2022-08-29 18:42:28 [verbose]: ⚠️ A newer release of osu! has been found (2022.821.0 → 2022.816.0).

No idea how this can happen.

**Reply by newtonGnu** [reply]
I reimported 2 beatmaps for testing (i doesn't work ingame because the game thinks the maps are already installed, so i have to redownload everything through web and import manually) and after a restart they worked fine.

After that, i managed to crash lazer when trying to watch a replay that apparently doesn't exist anymore. So even the scores are still visible, the replays aren't:
> 2022-08-30 08:12:23 [error]: An unhandled error has occurred.
> 2022-08-30 08:12:23 [error]: System.ArgumentNullException: Value cannot be null. (Parameter 'input')

Maybe there needs to be a check if the replay exists to handle the exception instead of crashing lazer
(i don't know if i should open an issue for this because it only happened on corrupted data)

full logs:
[crash_logs.zip](https://github.com/ppy/osu/files/9451156/crash_logs.zip)
